keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -507,6 +507,7 @@ class SqlDeliveryConfigRepository(
               .set(ENVIRONMENT_ARTIFACT_CONSTRAINT.JUDGED_AT, MySQLDSL.values(ENVIRONMENT_ARTIFACT_CONSTRAINT.JUDGED_AT))
               .set(ENVIRONMENT_ARTIFACT_CONSTRAINT.COMMENT, MySQLDSL.values(ENVIRONMENT_ARTIFACT_CONSTRAINT.COMMENT))
               .set(ENVIRONMENT_ARTIFACT_CONSTRAINT.ATTRIBUTES, MySQLDSL.values(ENVIRONMENT_ARTIFACT_CONSTRAINT.ATTRIBUTES))
+              .set(ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE, state.artifactReference)
```

Comment:
This was missing and so the column was never updated.

Reply:
Can you elaborate on why this needs to be set here? the setting is done at line 503, and the artifact reference should not be changed when updating the state.

Reply:
Because the `UNIQUE KEY` on the table is on `environment_uid`,`type` and `artifact_version` only, and with the other change below the record will load if `artifact_reference` is `NULL` (even though we actually have a reference for the artifact) and so will cause a duplicate exception here, so we want to take this opportunity to update the column with the right value.

Reply:
P.S. The alternative to this would have been to write a DB migration to fill in the missing reference, since it's mandatory in practice, which I feel should have been done in the original PR where this change was introduced.